src/Pagination/PaginateDirective.php

```diff
@@ -182,9 +183,12 @@ protected function optimalPaginationType(ResolveInfo $resolveInfo): PaginationTy
             return $type;
         }
 
+        // If it cached do not manipulate the paginator type.
```

Comment:
This comment just repeats what is already written in the code - but even more imprecise and foggy. Rather than this half-sentence, the comment should thoroughly explain the problem that is being solved.

Reply:
The result will be cached with a simple paginator type, however, the schema allowed other pagination types.
If pagination data is queried in the next request, the cache does not contain the data and throws an exception.